front/app/containers/Admin/projects/project/analysis/Tags/AutotaggingModal.tsx

```diff
@@ -1,43 +1,123 @@
-import { Box, Title, Button } from '@citizenlab/cl2-component-library';
-import { tagTypes } from 'api/analysis_tags/types';
-import React from 'react';
+import { Box, Title, Text, colors } from '@citizenlab/cl2-component-library';
+import { TagType } from 'api/analysis_tags/types';
+import React, { ReactNode } from 'react';
 
 import useLaunchAnalysisAutotagging from 'api/analysis_background_tasks/useLaunchAnalysisAutotagging';
 import { useParams } from 'react-router-dom';
 import { TagTypeColorMap } from './Tag';
+import styled from 'styled-components';
+
+const AutoTagOptionContainer = styled.div`
+  cursor: pointer;
+  background-color: ${colors.grey100};
+  border-radius: 3px;
+  display: flex;
+  align-items: center;
+  gap: 20px;
+  &:hover {
+    border: 1px black;
+    box-shadow: rgba(0, 0, 0, 0.02) 0px 1px 3px 0px,
+      rgba(27, 31, 35, 0.15) 0px 0px 0px 1px;
+  }
+  min-height: 90px;
+`;
+
+const AutoTagOption = ({
+  children,
+  tagType,
+  title,
+  onSelect,
+}: {
+  children: ReactNode;
+  tagType: TagType;
+  title: string;
+  onSelect: () => void;
+}) => {
+  return (
+    <AutoTagOptionContainer onClick={() => onSelect()}>
+      <Box
+        color={TagTypeColorMap[tagType].text}
+        background={TagTypeColorMap[tagType].background}
+        px="12px"
+        py="4px"
+        flex="0 0 160px"
+        ml="-14px"
+        boxShadow="rgba(50, 50, 105, 0.15) 0px 2px 5px 0px, rgba(0, 0, 0, 0.05) 0px 1px 1px 0px;"
+      >
+        {title}
+      </Box>
+      <Text px="20px">{children}</Text>
+    </AutoTagOptionContainer>
+  );
+};
 
 const AutotaggingModal = ({ onCloseModal }: { onCloseModal: () => void }) => {
   const { analysisId } = useParams() as { analysisId: string };
-  const { mutate: launchTagging, isLoading } = useLaunchAnalysisAutotagging();
+  const { mutate: launchTagging } = useLaunchAnalysisAutotagging();
```

Comment:
I still think it would be nice to show a loading state when this is happening before closing the modal.

Reply:
Will add